### PR TITLE
187810613 ruler add attribute

### PR DESCRIPTION
**Pull request by eireland**
Adds ability to add a new attribute in a collection from the table ruler menu.

### CODE REVIEW

**Review by kswenson — CHANGES_REQUESTED** [review]
Let's put this on hold until #1353 lands because @tealefristoe made changes in this vicinity as well.

File: v3/src/components/case-table/inspector-panel/ruler-menu-list.tsx
Comment (tealefristoe) [inline v3/src/components/case-table/inspector-panel/ruler-menu-list.tsx:49]:
```suggestion
            {t("DG.Inspector.newAttribute", { vars: [collection.title] })}
```

File: v3/src/components/case-table/collection-table.tsx
Comment (tealefristoe) [inline v3/src/components/case-table/collection-table.tsx:130]: It would be better to figure out the index based on the id, rather than assuming it's the right-most column. That's a safe assumption right now but it could change in the future, in which case this would cause an obscure bug to solve.

**Review by tealefristoe — CHANGES_REQUESTED** [review]
This looks pretty good, but it's going to require a merge with main. This implementation starts editing the attribute title immediately when you create a new one, which is a good touch my implementation didn't include. I think you should be able to mostly use this implementation over mine, but `isDisabled` will have to be brought over from mine. This is just a one line change.

One thing that confuses me about this PR is the change to case card. The PT story makes no mention of the case card. And looking through the code, it looks like the `caseMetaData` is being prop drilled here, but it isn't ever being used for anything. What am I missing?

**Review by nstclair-cc — APPROVED** [review]
I focused on reviewing `table.spec.ts`. LGTM! Thanks for updating the specs in the UI of the test. 👍🏼

File: v3/src/components/case-card/case-card.v2.js
Comment (tealefristoe) [inline v3/src/components/case-card/case-card.v2.js:29]:
```suggestion
```

File: v3/src/components/case-card/case-card.tsx
Comment (tealefristoe) [inline v3/src/components/case-card/case-card.tsx:25]:
```suggestion
  const { data } = useDataSet()
```
You could also revert back to `useDataSetContext()`... I'm not sure if there's an advantage either way.

File: v3/src/components/case-table/collection-table.tsx
Comment (tealefristoe) [inline v3/src/components/case-table/collection-table.tsx:131]:
```suggestion
      const newAttrIdx = columns.findIndex(column => column.key === collectionTableModel.attrIdToEdit)
```

File: v3/src/components/case-table/inspector-panel/ruler-menu-list.tsx
Comment (tealefristoe) [inline v3/src/components/case-table/inspector-panel/ruler-menu-list.tsx:10]: Remove line 5, `import { ICollectionModel } from "../../../models/data/collection"`.

**Review by tealefristoe — APPROVED** [review]
Still a few more things to clean up, but after that you should be good to go 👍

File: v3/src/components/case-card/case-card.v2.test.tsx
Comment (kswenson) [inline v3/src/components/case-card/case-card.v2.test.tsx:10]: These `metadata` imports and properties should be removed from the tests as well at some point.

**Pull request merged**
